### PR TITLE
task: Add a 'users' page to display all users

### PR DESCRIPTION
**Pull request by mwhitman189**
Add a page that displays all users in the database.

The 'UserList' component contains the styling, and the page itself resides in the 'pages' directory. The 'users' page simply imports the 'UserList' component.

(UserList component flow):
1. The class-based component queries the 'users' db table upon mounting
2. The component state is set in the fetch callback
3. The component iterates through the users, rendering a 'name-plate' div for each

Resolves: #2

## Self-check

- [ ] Reviewed files changed tab
- [ ] Tested locally for regression
- [ ] Added unit tests (where applicable)

### CODE REVIEW

File: client/components/userList.js [hunk 1]
```diff
@@ -17,16 +13,19 @@ class UserList extends Component {
...
-            .then(data => this.setState({
-                users: data.user.map(user => {
-                    return {
-                        username: user.username,
-                        nativeLanguage: user.nativeLanguage,
-                        points: user.points,
-                        rank: user.rank
-                    }
+            .then(data => {
+                console.log(data)
```
Comment (eliotc1986) [inline client/components/userList.js:17]: @mwhitman189 console.log

File: client/components/userList.js [hunk 2]
```diff
@@ -17,16 +13,19 @@ class UserList extends Component {
...
+                this.setState({
+                    users: data.user.map(user => {
```
Comment (eliotc1986) [inline client/components/userList.js:19]: @mwhitman189 I think it's better to do the mapping operation outside of set state.

File: client/components/userList.js [hunk 3]
```diff
@@ -17,16 +13,19 @@ class UserList extends Component {
...
+                        return {
+                            username: user.username,
+                            nativeLanguage: user.nativeLanguage,
+                            points: user.points,
+                            rank: user.rank
+                        }
+                    })
```
Comment (eliotc1986) [inline client/components/userList.js:26]: You can use destructuring to clean this code up a bit.

```suggestion
                    users: data.user.map(({ id, username, nativeLanguage, points, rank }) => {
                        return {
                            id,
                            username,
                            nativeLanguage,
                            points,
                            rank
                        }
                    })
```

File: client/components/userList.js [hunk 4]
```diff
@@ -17,16 +13,19 @@ class UserList extends Component {
...
                 })
-            }))
+            })
```
Comment (eliotc1986) [inline client/components/userList.js:28]: @mwhitman189 

-probably better to use pluralized user/users for your data structure.
- if you don't need to filter anything out the you could probably just do `users: data.users`

File: client/components/userList.js [hunk 5]
```diff
@@ -36,7 +35,28 @@ class UserList extends Component {
...
-                    return <UserPlate className="user-plate" key={user.username}>{user.username}</UserPlate>
+                    return (
+                        <>
+                            <div className="user-plate" key={user.username}>
+                                <div className="user-name bold">{user.username}</div>
+                            </div>
+                            <style jsx>{`
+                                .user-plate {
+                                    color: ${theme.dtColors.text};
+                                    background: ${theme.ltColors.link};
```
Comment (eliotc1986) [inline client/components/userList.js:46]: [Nit]: prefer to use "background-color" instead of shorthand "background"

File: client/pages/_app.js [hunk 6]
```diff
@@ -20,6 +20,10 @@ function App({ Component, pageProps }) {
         a:hover {
           color: ${pageProps.isDarkTheme ? theme.dtColors.linkHover : theme.ltColors.linkHover}
         }
+
+        .bold {
+          font-weight: bold;
```
Comment (eliotc1986) [inline client/pages/_app.js:25]: @mwhitman189 nit: unless it's a utility class that won't change largely, better to use semantic class names.

File: client/pages/users/users.js [hunk 7]
```diff
@@ -15,4 +15,5 @@ export default function Profile(props) {
             <UserList />
         </>
     )
-}
\ No newline at end of file
+}
```
Comment (eliotc1986) [inline client/pages/users/users.js:18]: @mwhitman189 get rid of props argument if it's not being used.
probably can use shorthand like this:

```
const funciton = () => (
   <Mycomponent ... />
)
```

**Review by eliotc1986 — COMMENTED** [review]
Thanks, I think you can clean up your code a bit by making use of destructuring and implict return of arrow functions.

File: client/pages/users/users.js (hunk 7 above)
Comment (mwhitman189) [reply client/pages/users/users.js:18]: Thanks for the suggestion!

 I will remove props.
There are actually <Head> and <h1> elements, so I think the explicit return is necessary.

**Review by mwhitman189 — COMMENTED** [review]
@eliotc1986 , I have addressed your recommendations, thanks

**Pull request merged**